docs/SRS-Volere/SRS.tex

```diff
@@ -132,6 +133,64 @@ \subsection{Individual Product Use Cases (PUC's)}
 \lips
 
 \section{Functional Requirements}
+\subsection{Formal Definitions}
+\begin{itemize}
+  \item Let \(U = \{u_1, u_2, \cdots, u_i\}\) be the set of users in the system.
+  \item Let \(R = \{r_1, r_2, \cdots, r_r\}\) be the set of roles (e.g. manager,
+  contractor, subcontractor)
+  \item Let \(A = \{a_1, a_2, \cdots, a_s\}\) be the set of actions (e.g.
+    upload, authenticate, sign, view)
+  \item Let \(C = \{c_1, c_2, \cdots, c_t\}\) be the set of compliance
+    requirements (e.g. safety training, hazard assesments)
+  \item Let \(I = \{l_1, l_2, \cdots, l_q\}\) be the set of allowed locations.
+  \item Let \(loc(u_i)\) represent the current location of user \(u_i\).
+  \item Let \(D = \{d_1, d_2, \cdots, d_n\}\) be the set of documents.
+  \item Let \(hasRole(u_i, r_k)\) represent that user \(u_i \in U\) has a role
+    \(r_k \in R\).
+  \item Let \(notify(u_m, u_i\) represent a notification from user \(u_m \in U\)
+    to user \(u_i \in U\).
+  \item Let \(manages(u_m, u_i\) represent that user \(u_m \in U\)
+    manages user \(u_i \in U\).
+  \item Let \(permitted(r_k, a_s, d_j\) represent that a user with role \(r_k \in R\)
+    is permitted to perform action \(a_s \in A\) on document \(d_j \in D\)
+  \item Let \(requires(a_s, c_t\) represent that action \(a_s \in A\) requires
+    compliance requirement \(c_t \in C\)
+  \item Let \(T(u_i) \subset C\) represent the set of completed training for a
+    user \(u_i \in U\).
+  \item Let \(assoc(d_j, u_i)\) denote that document \(d_j\) is associated with
+    user \(u_i\).
+  \item Let \(upload(u_i, d_j)\) represent the action of user \(u_i \in U\)
+    uploading document \(d_j \in D\)
+\end{itemize}
+\subsection{Formal Expressions}
+\begin{enumerate}
+  \item \(\forall u_i \in U, \forall d_j \in D, \forall r_k \in R, \forall a_s
+    \in A: \)
+    \begin{align*}
+      hasRole(u_i, r_k) \land permitted(r_k, a_s, d_j) \implies performAction(u_i, a_s, d_j)
+    \end{align*}
+  \item \(\forall u_i \in U, \forall c_t \in C:\)
+    \begin{align*}
+      (requires(a_s, c_t) \land a_s = sign \implies (c_t \in T(u_i)) \implies
+      performAction(u_i, a_s, d_j))
+    \end{align*}
+  \item \(\forall u_i \in U, \forall d_j \in D:\)
+    \begin{align*}
+      (loc(u_i) \in L) \implies upload(u_i, d_j)
```

Comment:
Should this also include how the user must be permitted to upload actions to upload the document?

Reply:
I guess like 
```suggestion
      (loc(u_i) \in L \land hasRole(u_i, r_k) \land a_s = upload \land permitted(r_k, a_s. d_j)) \implies upload(u_i, d_j)
```

or are we not checking permissions and just letting anybody upload?

Reply:
So that line is saying a user ui has role rk and the action is the action as  is upload and the role rk is permitted to perform action as, that that means their are able to upload document dj. Should that capture permissions? My thinking is that the set of roles R will have the permissions associated with rk defined

Reply:
Yeah I think that would cover the permissions

Reply:
I am approving so that we can get feedback on this tomorrow, but maybe we can come back to modify this later